### PR TITLE
Drop parent_fqdn in favor of deriving from foreman_url

### PR DESCRIPTION
**Pull request by ehelms**
I wanted to continue working toward the number of things a user has to specify to get things working and in this case the number of things a user has to specify multiple times, via different parameters. If we take https://github.com/theforeman/forklift/blob/master/pipelines/install/04-install_proxy.yaml#L11 as an example, the user effectively specifies the Foreman server hostname 4 times in the most basic of commands to install a content proxy.

### CODE REVIEW

File: manifests/init.pp
Comment (ekohl) [inline manifests/init.pp:185]: The Foreman URL can really be a URL and include a path. I think the safest would be to add a function to extract the hostname from a URL. https://github.com/voxpupuli/puppet-extlib/blob/master/lib/puppet/functions/extlib/dump_args.rb is probably the best example of a minimal function. Then use Ruby's URL parsing to do so.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: manifests/init.pp
Comment (ehelms) [reply manifests/init.pp:185]: Hah. I almost went that route. Then said... let me try the 'dumb' route and save myself time just in case :)

Comment (evgeni) [reply manifests/init.pp:185]: I wonder if that's actually true for Katello deployments, or if we hardcore paths like /rhsm on clients so that you can't deploy it in a "folder".

Comment (ekohl) [reply manifests/init.pp:185]: Still, writing your own parsing is a likely a source of problems. For example, even if we only support `/`, the current code doesn't deal with `https://foreman.example.com/` which is a valid input.

**Comment by ehelms** [comment]
Now with a function for deriving the hostname by parsing the URL. Admittedly, I couldn't come up with a better name than `hostname` for the function. Open to suggestions if others don't like it.

File: lib/puppet/functions/foreman_proxy_content/hostname.rb
Comment (ekohl) [inline lib/puppet/functions/foreman_proxy_content/hostname.rb:12]:
```suggestion
    return_type 'Stdlib::Host'
```

File: spec/functions/foreman_proxy_content/hostname_spec.rb
Comment (ekohl) [inline spec/functions/foreman_proxy_content/hostname_spec.rb:11]: Looks like you're using tabs while everywhere else we use spaces.

**Review by ekohl — COMMENTED** [review]
`url2hostname` perhaps?

File: lib/puppet/functions/foreman_proxy_content/hostname_from_url.rb
Comment (ekohl) [inline lib/puppet/functions/foreman_proxy_content/hostname_from_url.rb:6]: Functions need to be called with their namespace
```suggestion
#   $hostname = foreman_proxy_content::hostname_from_url('https://test.example.com')
```

File: lib/puppet/functions/foreman_proxy_content/hostname_from_url.rb
Comment (ekohl) [inline lib/puppet/functions/foreman_proxy_content/hostname_from_url.rb:17]: Interesting that https://ruby-doc.org/stdlib-2.7.2/libdoc/uri/rdoc/URI.html makes no mention of `hostname`. There is only host.

**Review by ekohl — CHANGES_REQUESTED** [review]
Looking at the URI docs and RFC I think we should use `host`. `hostname` implies it's a name, but technically an IP is also valid. Yes, I know that `ip:` is rare in `subjectAltName`, but let's follow the standard. We can also handle that fine in the other code.

File: lib/puppet/functions/foreman_proxy_content/host_from_url.rb
Comment (ekohl) [inline lib/puppet/functions/foreman_proxy_content/host_from_url.rb:6]: This still needs the namespace
```suggestion
#   $host = foreman_proxy_content::host_from_url('https://test.example.com')
```

File: manifests/init.pp
Comment (ekohl) [inline manifests/init.pp:185]: Let's make this host too
```suggestion
  $foreman_host = foreman_proxy_content::host_from_url($foreman_url)
```

**Review by ekohl — COMMENTED** [review]
2 small nits, otherwise :+1:

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
This is backwards incompatible and we will need to handle this in forklift where we pass it in our CI. Do you want to wait for that PR to be ready before we merge it?

**Pull request merged**
